### PR TITLE
Refactor HashSetCasperTestNode to allow any F

### PR DESCRIPTION
**Pull request by birchmd**
## Overview
This enables setting up a network of nodes which use `Task` and thus can act in parallel. It also removes the code duplication present in `helper.CasperEffect` vs `helper.HashSetCasperTestNode`.

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
NO-TICKET

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You assigned one person to review this PR

### If you are not a member of the core development team, please confirm:
- [ ] You signed the commit. Merging requires a signature. Please see the (developer wiki)[https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain] for instructions.
- [ ] Your GitHub account is also an account with (Travis CI)[https://travis-ci.org]. Unit tests will not run on your PR unless you have an account with Travis. Merge requires passed unit tests.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala [hunk 1]
```diff
@@ -12,37 +13,55 @@ import scala.concurrent.duration.FiniteDuration
 import scala.collection.mutable
 import coop.rchain.comm.transport._
 
-class TransportLayerTestImpl[F[_]: Monad: Capture](
+class TransportLayerTestImpl[F[_]: Monad](
     identity: PeerNode,
-    val msgQueues: collection.Map[PeerNode, mutable.Queue[Protocol]])
+    val msgQueues: Map[PeerNode, Ref[F, mutable.Queue[Protocol]]])
```
Comment (goral09) [inline casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala]: If you don't plan to modify the `msgQueues` from the outside of `TransportlayerImpl` then you shouldn't need `Ref`. Am I missing something?

Comment (birchmd) [reply casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala]: The whole point of the reactor is to enable concurrent unit tests involving a network of nodes (in the same style as as existing unit tests involving race conditions on a single node). There is one set of queues for the whole network, so there could be concurrent modifications as one node sends a message at the same time as the recipient receives a different message.

Comment (goral09) [reply casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala]: I didn't notice that in the code. Thanks for explaining.

Comment (birchmd) [reply casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala]: To be fair, I haven't pushed an example of using it in that way yet. I'm trying to write a test that reproduces Rhol-722, but have not been successful yet.

**Pull request merged**